File: TA/Backend/SoftRhRMI_Servidor/src/pe/edu/pucp/softrh/rmi/interfaces/AdministradorBO.java
package pe.edu.pucp.softrh.rmi.interfaces;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.usuarios.model.Administrador;

public interface AdministradorBO extends Remote{

    Integer insertar(String dni, String nombres, String apellidos, String correo, String contrasenha, Date fechaCreacion) throws RemoteException;

    Integer modificar(Integer idUsuario, String dni, String nombres, String apellidos, String correo, String contrasenha, Date fechaCreacion) throws RemoteException;

    Integer eliminar(Integer idAdministrador) throws RemoteException;

    ArrayList<Administrador> listarTodos() throws RemoteException;

    Administrador obtenerPorId(Integer idAdministrador) throws RemoteException;

    ArrayList<Administrador> listarPorDniNombre(String cadena) throws RemoteException;

}

File: TA/Frontend/RHStoreWS/RHStoreUsuariosBO/AdministradorBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreUsuariosBO
{
	public class AdministradorBO : BaseBO
	{
        public int insertar(string dni, string nombres, string apellidos, string correo, string contrasenha, DateTime fechaCreacion)
        {
            return AdministradorWS.insertarAdministrador(dni, nombres, apellidos, correo, contrasenha, fechaCreacion);
        }

        public int modificar(int idUsuario, string dni, string nombres, string apellidos, string correo, string contrasenha, DateTime fechaCreacion)
        {
            return AdministradorWS.modificarAdministrador(idUsuario, dni, nombres, apellidos, correo, contrasenha, fechaCreacion);
        }

        public int eliminar(int idAdministrador)
        {
            return AdministradorWS.eliminarAdministrador(idAdministrador);
        }

        public BindingList<administrador> listarTodos()
        {

            try
            {
                List<administrador> administradores = AdministradorWS.listarAdministradoresTodos()?.ToList();
                return new BindingList<administrador>(administradores ?? new List<administrador>());
            }
            catch (Exception ex)
            {
                // Manejar la excepción (registrarla o notificar al usuario)
                return new BindingList<administrador>(new List<administrador>());
            }

        }

        public administrador obtenerPorId(int idAdministrador)
        {
            return AdministradorWS.obtenerAdministradorPorId(idAdministrador);
        }

		public BindingList<administrador> listarPorDniNombre(string cadena)
		{
            try
            {
                List<administrador> administradores = AdministradorWS.listarAdministradoresPorDniNombre(cadena)?.ToList();
                return new BindingList<administrador>(administradores ?? new List<administrador>());
            }
            catch (Exception ex)
            {
                // Manejar la excepción (registrarla o notificar al usuario)
                return new BindingList<administrador>(new List<administrador>());
            }
        }
	}
}
